topydo/lib/PrettyPrinterFilter.py

```diff
@@ -125,3 +158,115 @@ def filter(self, p_todo_str, _):
                                 p_todo_str)
 
         return p_todo_str
+
+
+class PrettyPrinterBasicPriorityFilter(PrettyPrinterFilter):
+
+    """ Strips the bracked from the priority of the todo item. Add a (blank)
+        space if not priority is specified (to maintain printing alignment)
+    """
+
+    def __init__(self, p_replacement=" "):
+        super(PrettyPrinterBasicPriorityFilter, self).__init__()
+        self.p_replacement = p_replacement
+
+    def filter(self, p_todo_str, _):
+        """ Find the priority """
+        matches = re.search('^(?P<id>\| *\w+\| )?(\((?P<pri>[A-Z])\) )?', p_todo_str)
+        # we are looking for the id string and priority at the beginning of the
+        # line, in the form of:
+        #   | 48| (V) ...
+        #   |h2j| (C) ...
+        if matches:
+            if matches.group('id'):
+                if matches.group('pri'):
+                    """ If we have a priority """
+                    return(matches.group('id') + matches.group('pri') + ' ' + p_todo_str[matches.span()[1]:])
+                else:
+                    return(matches.group('id') + self.p_replacement + ' ' + p_todo_str[matches.span()[1]:])
+            else:
+                if matches.group('pri'):
+                    """ If we have a priority """
+                    return(matches.group('pri') + ' ' + p_todo_str[matches.span()[1]:])
+                else:
+                    return(self.p_replacement + ' ' + p_todo_str[matches.span()[1]:])
+        else:
+            """ No match found """
+            return(self.p_replacement + ' ' + p_todo_str)
+
+
+class PrettyPrinterHumanDatesFilter(PrettyPrinterFilter):
+
+    """ Turns dates to human readable versions. """
+
+    def __init__(self):
+        super(PrettyPrinterHumanDatesFilter, self).__init__()
+
+    def humanize_from_match(self, matchgroup):
+        """
+        Takes a match group, compares it to 'now', and then returns a 'human
+        readable' version of the difference.
+
+        Assumes the matchgroup has matches named 'year', 'month', and 'day'.
+        """
+        linedate = arrow.now()  # set the time to 'now' so the comparisions work better
+        linedate = linedate.replace(year=int(matchgroup.group('year')))
+        linedate = linedate.replace(month=int(matchgroup.group('month')))
+        linedate = linedate.replace(day=int(matchgroup.group('day')))
+        delta = linedate.humanize()
```

Comment:
Fixed in 0130560. New output:

```
|354| Z +test (just now, due just now, threshold of 9 months ago)
|353| Z +test (just now, threshold in a year)
```

I think the grammar on due and added dates is working as arrow provides, at least in English. If the program is translated, that could bring up other issues.